extensions/amp-analytics/0.1/events.js

```diff
@@ -1184,22 +1189,35 @@ export class VisibilityTracker extends EventTracker {
   }
 
   /**
+   * @param {string} reportWhenSpec
    * @return {!Promise}
    */
-  createReportReadyPromise_() {
+  createReportReadyPromise_(reportWhenSpec) {
     const viewer = this.root.getViewer();
 
-    if (!viewer.isVisible()) {
-      return Promise.resolve();
-    }
-
-    return new Promise(resolve => {
-      viewer.onVisibilityChanged(() => {
+    switch (reportWhenSpec) {
+      case 'hidden':
```

Comment:
nit: instead of having a switch block and inlining function here? Can we define those functions first and return the already defined function instead. This can prevent the block from growing too big which makes it hard to read.